common/src/api/external/mod.rs

```diff
@@ -629,12 +629,19 @@ pub struct IdentityMetadataUpdateParams {
 )]
 #[serde(rename_all = "snake_case")]
 pub enum InstanceState {
-    Creating, // TODO-polish: paper over Creating in the API with Starting?
+    /// An Instance record exists but the Instance is currently being created on the Sled.
+    Creating,
+    /// The Instance and its resources are being brought up.
+    Provisioning,
+    /// The Instance and all its resources have been allocated.
+    Provisioned,
+    /// The Instance is currently starting up.
     Starting,
+    /// The Instance is currently running.
     Running,
-    /// Implied that a transition to "Stopped" is imminent.
+    /// The Instance has been requested to stop and a transition to "Stopped" is imminent.
     Stopping,
-    /// The instance is currently stopped.
+    /// The instance is currently stopped and its resources deallocated.
```

Comment:
I'm not sure this adds clarity in its role as customer documentation. It may be unclear and it may be imprecise in that disk resources certainly are not "deallocated".  I'd suggest leaving the docs as they were.